feature-libs/quote/components/details/edit/quote-details-edit.component.spec.ts

```diff
@@ -0,0 +1,267 @@
+      expect,
+      htmlElem,
+      'button.btn-secondary'
+    );
+
+    CommonQuoteTestUtilsService.expectElementToContainText(
+      expect,
+      htmlElem,
+      'button.btn-secondary',
+      'common.save'
+    );
+  });
+
+  it('should handle cancel action', () => {
+    const cancelButton = CommonQuoteTestUtilsService.getNativeElement(
+      debugElement,
+      'button.btn-tertiary'
+    );
+    cancelButton.click();
+    expect(component.cancelCard.emit).toHaveBeenCalled();
+    expect(component.cancelCard.emit).toHaveBeenCalledWith(false);
+  });
+
+  it('should handle edit action', () => {
+    const newTextForTitle1: any = 'New title for name';
+    const newTextForTitle2: any = 'Here could be found a long description';
+    component.editForm.get('name')?.setValue(newTextForTitle1);
+    component.editForm.get('name').markAsTouched();
+    component.editForm.get('description')?.setValue(newTextForTitle2);
+    component.editForm.get('description').markAsTouched();
+    fixture.detectChanges();
+    const saveButton = CommonQuoteTestUtilsService.getNativeElement(
+      debugElement,
+      'button.btn-secondary'
+    );
+    saveButton.click();
+    expect(component.editCard.emit).toHaveBeenCalled();
+    let arg: any = (component.editCard.emit as any).calls.mostRecent().args[0];
+    expect(arg.editMode).toEqual(false);
+    expect(arg.name).toEqual(newTextForTitle1);
+    expect(arg.description).toEqual(newTextForTitle2);
+  });
+
+  describe('setFormControlName', () => {
+    it('should return form control name converted to lower case', () => {
+      let name = 'TeSt';
+      let convertedName = name.toLocaleLowerCase();
+      expect(component['setFormControlName'](name)).toBe(convertedName);
+
+      name = 'TEST';
+      convertedName = name.toLocaleLowerCase();
+      expect(component['setFormControlName'](name)).toBe(convertedName);
+    });
+  });
+
+  describe('getCharactersLeft', () => {
+    function setValue(formControlName: string, value: any): void {
+      component.editForm.get(formControlName)?.setValue(value);
+      component.editForm.get(formControlName).markAsTouched();
+      fixture.detectChanges();
+    }
+
+    it('should calculate left characters', () => {
+      const formControlName = 'description';
+      setValue(formControlName, 'New title for name');
+
+      let charactersLeft =
+        component.content.paragraphs[1].charactersLimit -
+        component.editForm.get(formControlName)?.value?.length;
+      expect(
+        component['getCharactersLeft'](
+          component.content.paragraphs[1].title,
+          component.content.paragraphs[1].charactersLimit
+        )
+      ).toBe(charactersLeft);
+
+      charactersLeft =
+        component.content.paragraphs[1].charactersLimit -
+        component.editForm.get(formControlName)?.value?.length;
+      expect(
+        component['getCharactersLeft'](
+          component.content.paragraphs[1].title,
```

Comment:
Strict issue, also please fix the following ones

Reply:
Done